bench-vortex/src/bin/tpch.rs

```diff
@@ -204,133 +310,133 @@ async fn bench_main(
+                            time: fastest_result,
+                            format,
+                            dataset: "tpch".to_owned(),
+                        });
 
-            let mut fastest_result = Duration::from_millis(u64::MAX);
-            for i in 0..iterations {
-                let start = Instant::now();
-                let (row_count, plan) = run_tpch_query(&ctx, &sql_queries, query_idx).await;
-                let elapsed = start.elapsed();
-                fastest_result = fastest_result.min(elapsed);
-
-                if i == 0 {
-                    row_counts.push((query_idx, format, row_count));
-                    // gather metrics
-                    for (idx, m) in VortexMetricsFinder::find_all(plan.as_ref())
-                        .into_iter()
-                        .enumerate()
-                    {
-                        metrics.merge_all_with_label(
-                            m,
-                            &[
-                                Label::new("query_idx", query_idx.to_string()),
-                                Label::new("vortex_exec_idx", idx.to_string()),
-                            ],
-                        );
+                        progress.inc(1);
                     }
-                    plans.push((query_idx, plan));
-                }
-            }
 
-            let storage = match url.scheme() {
-                "s3" => "s3",
-                "gcs" => "gcs",
-                "file" => "nvme",
-                otherwise => {
-                    warn!("unknown URL scheme: {}", otherwise);
-                    return ExitCode::FAILURE;
+                    if export_spans {
+                        if let Err(e) = export_plan_spans(format, &plans).await {
+                            warn!("failed to export spans {e}");
+                        }
+                    }
                 }
-            }
-            .to_owned();
-
-            measurements.push(QueryMeasurement {
-                query_idx,
-                engine: Engine::DataFusion,
-                storage,
-                time: fastest_result,
-                format,
-                dataset: "tpch".to_owned(),
-            });
+                Engine::DuckDB => {
+                    let duckdb_executable = ddb::executable_path(duckdb_path);
+
+                    for (query_idx, sql_queries) in tpch_queries.clone() {
+                        let fastest_run = benchmark_duckdb_query(
+                            query_idx,
+                            &sql_queries,
+                            iterations,
+                            format,
+                            &url,
+                            &duckdb_executable,
+                        );
 
-            progress.inc(1);
-        }
+                        let storage = match url.scheme() {
+                            "s3" => "s3",
+                            "gcs" => "gcs",
+                            "file" => "nvme",
```

Comment:
maybe a try_from or something else

Reply:
This code was only moved, not introduced in this PR.